analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantInheritanceList.cs

```diff
@@ -49,76 +50,54 @@ public sealed class RedundantInheritanceList : SonarDiagnosticAnalyzer
+                        ReportRedundantBaseType(c, enumDeclaration, KnownType.System_Int32, MessageEnum);
+                        break;
+                    case InterfaceDeclarationSyntax interfaceDeclaration:
+                        ReportRedundantInterfaces(c, interfaceDeclaration);
+                        break;
+                    case TypeDeclarationSyntax nonInterfaceDeclaration:
+                        ReportRedundantBaseType(c, nonInterfaceDeclaration, KnownType.System_Object, MessageObjectBase);
+                        ReportRedundantInterfaces(c, nonInterfaceDeclaration);
+                        break;
+                    default:
+                        throw new InvalidOperationException($"node type {c.Node?.Kind()} not supported.");
+                }
+            },
+            SyntaxKind.EnumDeclaration,
+            SyntaxKind.InterfaceDeclaration,
+            SyntaxKind.ClassDeclaration,
+            SyntaxKind.StructDeclaration,
+            SyntaxKindEx.RecordClassDeclaration,
+            SyntaxKindEx.RecordStructDeclaration);
+
+        private static void ReportRedundantBaseType(SyntaxNodeAnalysisContext context, BaseTypeDeclarationSyntax typeDeclaration, KnownType redundantType, string message)
+        {
             var baseTypeSyntax = typeDeclaration.BaseList.Types.First().Type;
-            if (!(context.SemanticModel.GetSymbolInfo(baseTypeSyntax).Symbol is ITypeSymbol baseTypeSymbol))
+            if (context.SemanticModel.GetSymbolInfo(baseTypeSyntax).Symbol is not ITypeSymbol baseTypeSymbol)
             {
                 return;
             }
-
-            if (baseTypeSymbol.Is(KnownType.System_Object))
+            if (baseTypeSymbol.Is(redundantType))
             {
                 var location = GetLocationWithToken(baseTypeSyntax, typeDeclaration.BaseList.Types);
-                context.ReportIssue(
-                    Diagnostic.Create(Rule, location,
-                                      ImmutableDictionary<string, string>.Empty.Add(RedundantIndexKey, "0"),
-                                      MessageObjectBase));
-            }
-
-            ReportRedundantInterfaces(context, typeDeclaration);
-        }
-
-        private static void CheckInterface(SyntaxNodeAnalysisContext context)
-        {
-            var interfaceDeclaration = (InterfaceDeclarationSyntax)context.Node;
-            if (IsBaseListNullOrEmpty(interfaceDeclaration))
-            {
-                return;
-            }
-
-            ReportRedundantInterfaces(context, interfaceDeclaration);
-        }
-
-        private static void CheckEnum(SyntaxNodeAnalysisContext context)
-        {
-            var enumDeclaration = (EnumDeclarationSyntax)context.Node;
-            if (IsBaseListNullOrEmpty(enumDeclaration))
-            {
-                return;
+                context.ReportIssue(Diagnostic.Create(Rule, location, DiagnosticsProperties(redundantIndex: 0), message));
```

Comment:
I would like to keep it. If you look at the other arguments, you see `Rule`, `location`, and `message`. But what is this `0` all about? I in general use argument names for number literals, because it helps the reader to understand what that argument is about.

Reply:
That's obvious from the signature of that method

Reply:
> That's obvious from the signature of that method

That doesn't help on the call side. You have to peek definition or something like that to reveal the parameter name. That's painful and not always possible (e.g. GH review).

Reply:
GH website isn't the best place to read code effectively. We should not please that tool. It's easy enough in the IDE